### PR TITLE
Docker registry

### PR DESCRIPTION
**Pull request by jdoliner**
This PR adds the following thing:

- `make launch-registry` and `make clean-launch-registry` which deploys a registry in the k8s cluster.
- `create-pipeline --push-images` which will push the images referenced in the pipeline spec to the k8s registry

TODO:
- make the registry assets part of `pachctl deploy`
- make it possible to use the k8s add-on registry.

### CODE REVIEW

**Comment by dwhitena** [comment]
+1 for this functionality.  A huge improvement to the workflow.  If no one gets to it before me, I'm going to look through this code one night.  Thanks.

**Comment by sjezewski** [comment]
This is way cool!

I'd like us to get us in the habit of framing PRs from a user perspective (when applicable of course). If I was to take a swing at this one, I'd say something like:

"As a user ... having a private registry that pachctl updates when I create/update a pipeline allows me to seamlessly update my pipelines when my code is under active development"

Is that about right?

**Comment by jdoliner** [comment]
@sjezewski that's a reasonable framing of this PR. I don't think that's really going to work generally for PRs though. Just looking at the PRs we have open right now this is the only one that you can really frame like that. It probably fits better in the design stages of a new feature, which this PR should have included, my bad on that one.

**Comment by sjezewski** [comment]
That makes sense. Especially since we want to get PRs to be in as small / digestible chunks as we can. And those won't always correspond to 'user facing things' in and of themselves.

One nice side effect of framing it for a user is that it highlights how much work we're doing is directly user facing vs technical debt etc.

And by 'design phase' you mean an outline/discussion on the GH issue right? I know sometimes we use gdocs which makes sense for a discussion, but since our code (and GH) is our interface to our users, I think it's worth getting in the habit of framing things here on GH (even if only in summary).

File: src/server/pkg/deploy/assets/assets.go [hunk 1]
```diff
@@ -506,6 +508,105 @@ func InitJob(version string) *extensions.Job {
 	}
 }
 
+func RegistryRc() *api.ReplicationController {
```
Comment (msteffen) [inline src/server/pkg/deploy/assets/assets.go]: Just to aid my own understanding, are this and etc/kube/registry-rc.yaml the same? If so, why define them both?

**Comment by jdoliner** [comment]
The `make launch-registry` stuff was the first initial way I was spinning up the registry, will remove it now that the registry is part of deployment.

Vendored deps are changing because I added in `go-dockerclient` and it had a few dependencies that needed to be updated.

TODOS for this pr:

- [ ] remove yaml manifests and `make launch-registry` target.
- [ ] add storage for registry
- [ ] regen + extend docs

**Comment by jdoliner** [comment]
I decided not to tackle adding storage to the registry in this PR. Getting that to work correctly on every provider would require a lot of work. And it's unclear exactly what the right way is on each provider since there are a few options. More over I don't think that productionized registry inside the cluster are really the best solution, people are probably better off using things like gcr if they're on gce or EC2 container registry for amazon.

**Comment by jdoliner** [comment]
This is ready for review.

**Comment by msteffen** [comment]
Maybe down the line it would make sense to have a per-cluster config that points to the registry used for pipelines in that cluster? Then pachctl --push-image could just get this from the cluster.

Probably outside the scope of this PR, but just an idea

**Comment by jdoliner** [comment]
@msteffen 100% agree on that, we recently added a config file for pachctl stuff that would be a great place to put that. Gonna consider it outside the scope though.

**Comment by msteffen** [comment]
cookie monster lgtm

**Pull request merged**